docs/sdk/authentication/session-sigs/get-session-sigs.md

```diff
@@ -3,112 +3,256 @@ sidebar_position: 2
+The function uses the session keypair generated by the `LitNodeClient` to invoke a callback function that signs the generated session key, creating an `AuthSig` that is scoped to specific capabilities.
 
-You can generate a `SessionSig` with the help of `capacityDelegationAuthSig` object in the following way:
+Using the `getSessionSigs` function, you can specify the abilities of your current session on the Lit network. This function is the simplest way to get session signatures, at the minimum only requiring an `ethers.Wallet` and the `LitNodeClient`. It will enable your `ethers.Wallet` to perform abilities to the resources you specify in the `AuthSig`.
 
-```javascript
-const { LitNodeClient } = require('@lit-protocol/lit-node-client');
-const { LitNetwork } = require('@lit-protocol/constants');
-const { LitAbility, LitActionResource, createSiweMessageWithRecaps } = require('@lit-protocol/auth-helpers');
-const { ethers } = require('ethers');
+## Prerequisites
 
-// Create a new ethers.js Wallet instance
-const wallet = new ethers.Wallet(process.env.YOUR_PRIVATE_KEY);
+Before continuing this guide, you should have an understanding of:
+- [Session Signatures](./intro)
+- [Lit Resources and Abilities](./resources-and-abilities.md)
 
-const sessionSigsFunction = async() => {
-  // Instantiate a LitNodeClient
-  const litNodeClient = new LitNodeClient({
-    litNetwork: LitNetwork.DatilDev,
-    debug: true,
-  });
-  await litNodeClient.connect();
+## `getSessionSigs`' Interface
+[Source code](https://github.com/LIT-Protocol/js-sdk/blob/master/packages/types/src/lib/interfaces.ts#L1057-L1131)
 
-  /*
-  * When the getSessionSigs function is called, it will generate a session key
-  * and sign it using a callback function. The authNeededCallback parameter
-  * in this function is optional. If you don't pass this callback,
-  * then the user will be prompted to authenticate with their wallet.
-  */
-  const authNeededCallback = async ({
+```ts
+interface CommonGetSessionSigsProps {
+  pkpPublicKey?: string;
+  expiration?: any;
+  chain?: Chain;
+  resourceAbilityRequests: LitResourceAbilityRequest[];
+  sessionCapabilityObject?: ISessionCapabilityObject;
+  switchChain?: boolean;
+  sessionKey?: SessionKeyPair;
+  capabilityAuthSigs?: AuthSig[];
+}
+
+interface GetSessionSigsProps
+  extends CommonGetSessionSigsProps,
+    LitActionSdkParams {
+  authNeededCallback: AuthCallback;
+}
+
+getSessionSigs = async (
+    params: GetSessionSigsProps
+  ): Promise<SessionSigsMap>
+
+interface AuthSig {
+  sig: any;
+  derivedVia: string;
+  signedMessage: string;
+  address: string;
+}
+
+type SessionSigsMap = Record<string, AuthSig>;
+```
```

Comment:
From Daryl's feedback, we shouldn't include links to the source code or have the source code included in the docs because this is hard to maintain whenever there's a change to the interface within the SDK. So, this section should be something like:

```
## `getSessionSigs` Interface

For details on this methods interface, please go [here](https://v6-api-doc-lit-js-sdk.vercel.app/classes/lit_node_client_src.LitNodeClientNodeJs.html#getSessionSigs).
```

Also open to other ideas of how this should look like if you have them

Reply:
And this would imply we remove the `Parameters` and `Return Value` sections, as this info should be in the SDK API docs. For this I think we should review the SDK API docs and if the information isn't better than what you have here, we should make a PR into the SDK to improve the docs

Reply:
I guess if Daryl doesn't recommend any source code (snippits or links) in the docs then we can only reference the API docs. I agree with just linking the API docs then.

I based my `Parameters` and `Return Value` sections off of the `Wrapped Keys` guides, with most of the information being directly from the SDK (so via the API docs). I think explaining parameters/returns in the docs is worthwhile even if the information is in the API docs as well, looks better and makes the concepts more understandable as well IMO. Could we get a third opinion on this?

Reply:
> I think explaining parameters/returns in the docs is worthwhile even if the information is in the API docs as well, looks better and makes the concepts more understandable as well IMO

I agree, but the problem Daryl raised is we now have to manage the information in both the docs and the SDK API docs, instead of just directing users to the API docs which will (in theory) always be kept up to date since the docs live next to the code

If the SDK API docs aren't covering what each parameter/return value is sufficiently, then we should improve those docs, not write them in two places

Reply:
Alright, I removed the sections from the PR. I don't think the SDK API Docs are very good, so I'll make a PR with better explanations